x-pack/plugins/ml/public/explorer/select_limit/select_limit.js

```diff
@@ -7,65 +7,110 @@
+  let limit = LIMIT_OPTIONS.find(opt => (opt.val === value));
+
+  // Default to 10 if supplied value doesn't map to one of the options.
+  if (limit === undefined) {
+    limit = LIMIT_OPTIONS[1];
+  }
+
+  return limit;
+}
+
+// This service will be populated by the corresponding angularjs based one.
+export const mlSelectLimitService = {
+  initialized: false,
+  state: null
+};
+
+class SelectLimit extends Component {
+  constructor(props) {
+    super(props);
 
-import { stateFactoryProvider } from 'plugins/ml/factories/state_factory';
-
-import template from './select_limit.html';
-import 'plugins/ml/components/controls/controls_select';
-
-import { uiModules } from 'ui/modules';
-const module = uiModules.get('apps/ml');
-
-module
-  .service('mlSelectLimitService', function (Private) {
-    const stateFactory = Private(stateFactoryProvider);
-    this.state = stateFactory('mlSelectLimit', {
-      limit: { display: '10', val: 10 }
-    });
-  })
-  .directive('mlSelectLimit', function (mlSelectLimitService) {
-    return {
-      restrict: 'E',
-      template,
-      link: function (scope, element) {
-        scope.limitOptions = [
-          { display: '5', val: 5 },
-          { display: '10', val: 10 },
-          { display: '25', val: 25 },
-          { display: '50', val: 50 }
-        ];
-
-        const limitState = mlSelectLimitService.state.get('limit');
-        const limitValue = _.get(limitState, 'val', 0);
-        let limitOption = scope.limitOptions.find(d => d.val === limitValue);
-        if (limitOption === undefined) {
-          // Attempt to set value in URL which doesn't map to one of the options.
-          limitOption = scope.limitOptions.find(d => d.val === 10);
-        }
-        scope.limit = limitOption;
-        mlSelectLimitService.state.set('limit', scope.limit);
-
-        scope.setLimit = function (limit) {
-          if (!_.isEqual(scope.limit, limit)) {
-            scope.limit = limit;
-            mlSelectLimitService.state.set('limit', scope.limit).changed();
-          }
-        };
-
-        function setLimit() {
-          scope.setLimit(mlSelectLimitService.state.get('limit'));
-        }
-
-        mlSelectLimitService.state.watch(setLimit);
-
-        element.on('$destroy', () => {
-          mlSelectLimitService.state.unwatch(setLimit);
-          scope.$destroy();
-        });
-      }
+    // Restore the limit from the state, or default to 10.
+    if (mlSelectLimitService.initialized) {
+      this.mlSelectLimitService = mlSelectLimitService;
+    }
+
+    this.state = {
+      valueDisplay: LIMIT_OPTIONS[1].display,
     };
-  });
+  }
+
+  componentDidMount() {
+    // set initial state from service if available
+    if (this.mlSelectLimitService !== undefined) {
+      const limitState = this.mlSelectLimitService.state.get('limit');
+      const limitValue = _.get(limitState, 'val', 10);
```

Comment:
same thing as above with only importing `get` from lodash

Reply:
Done in 85d1567592.